docs/index.md

```diff
@@ -662,6 +662,8 @@ describe('outer', function() {
 });
 ```
 
+Skipping a test within an "after each" or "after all" hook is deprecated and will throw an exception in a future version of Mocha. Use a return statement or other means to abort hook execution.
```

Comment:
Can't say I really understand the point of the second sentence here in this context.

Reply:
`this.skip()` aborts the execution of the hook, subsequent code will not be executed.

Reply:
This new behavior? My understanding was that `this.skip` marked the test as pending. Execution of the hook continues regardless as documented on L611.

Reply:
There is no new behavior, it's just a soft deprecation without any change in the logic.
So the current behavior (aborting execution, marking as pending) remains.